sdk/inc/azure/core/internal/az_http_internal.h

```diff
@@ -193,19 +198,39 @@ AZ_NODISCARD az_result az_http_request_init(
  * For instance, if url in request is `http://example.net?qp=1` and this function is called with
  * path equals to `test`, then request url will be updated to `http://example.net/test?qp=1`.
  *
+ * @remarks if \p is_url_encoded is set to false, before appending the path, it would be
+ * url-encoded.
  *
- * @param ref_request http request builder reference
- * @param path span to a path to be appended into url
- * @return AZ_NODISCARD az_http_request_append_path
+ *
+ * @param[out] ref_request http request reference
+ * @param[in] path span to a path to be appended into url
+ * @param[in] \p is_url_encoded boolean value that defines if the value to be appended is
+ * url-encoded or not.
+ * @return
+ *   - *`AZ_OK`* success.
+ *   - *`AZ_ERROR_INSUFFICIENT_SPAN_SIZE`* the `URL` would grow past the `max_url_size`, should
+ * the parameter get set.
+ *   - *`AZ_ERROR_ARG`*
+ *     - `ref_request` is _NULL_.
```

Comment:
doc nit:
```suggestion
 * @return An #az_result value indicating the result of the operation:
 *         - #AZ_OK if successful
 *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the \p path wouldn't fit into the URL buffer provided.
```

Reply:
Also remove `AZ_ERROR_ARG`. We shouldn't be returning that. `ref_request` being null would fail the precondition.